app/src/test/java/com/onboarding/calculator/CalculatorTest.java

```diff
@@ -0,0 +1,213 @@
+import static com.onboarding.calculator.util.ConstantsUtils.EMPTY_STRING;
+import static com.onboarding.calculator.util.ConstantsUtils.Error;
+import static com.onboarding.calculator.util.ConstantsUtils.MUL;
+import static com.onboarding.calculator.util.ConstantsUtils.ONE_STRING_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_CLEAN_A_VALUE_IN_SECOND_OPERAND_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_CLEAN_OPERATOR_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_MINUS_IN_OPERATOR_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.OPERATION_MINUS_IN_SECOND_OPERAND_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.SUB;
+import static com.onboarding.calculator.util.ConstantsUtils.THREE_STRING_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.TWO_STRING_TEST;
+import static com.onboarding.calculator.util.ConstantsUtils.ZERO_STRING_TEST;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+public class CalculatorTest {
+
+    private CalculatorContract.Model model;
+    private CalculatorContract.Presenter presenter;
+    private CalculatorContract.View view;
+
+    @Before
+    public void setup() {
+        model = new CalculatorModel();
+        view = mock(CalculatorContract.View.class);
+        presenter = new CalculatorPresenter(model, view);
+    }
+
+    @Test
+    public void setACorrectOperator() {
+        //Test
+        presenter.onOperatorButtonPressed(ADD);
+        //Assert
+        assertEquals(ADD, model.getLastModified());
+        verify(view).showOperator(model.getLastModified());
+        verify(view).operationViewUpdate(model.getLastModified());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void setValuesInFirstOperand() {
+        //Test
+        presenter.onCalculatorButtonPressed(ONE_STRING_TEST);
+        //Assert
+        assertEquals(ONE_STRING_TEST, model.getLastModified());
+        verify(view).showValues(model.getLastModified());
+        verify(view).operationViewUpdate(model.getLastModified());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void setValuesInSecondOperand() {
+        //Set
+        model.setValues(ONE_STRING_TEST);
+        model.setOperator(ADD);
+        //Test
+        presenter.onCalculatorButtonPressed(TWO_STRING_TEST);
+        //Assert
+        assertEquals(TWO_STRING_TEST, model.getLastModified());
+        verify(view).showValues(model.getLastModified());
+        verify(view).operationViewUpdate(model.getOperation());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void CleanAllTheOperation() {
+        presenter.deleteAll();
+        assertEquals(EMPTY_STRING, model.getOperation());
```

Comment:
the same at the previous commment, check not only the operation, intead check if all the field are empty

Reply:
Solved in a call!